skrub/_joiner.py

```diff
@@ -45,66 +130,70 @@ class Joiner(TransformerMixin, BaseEstimator):
     suffix : str, default=""
         Suffix to append to the ``aux_table``'s column names. You can use it
         to avoid duplicate column names in the join.
-    match_score : float, default=0
-        Distance score between the closest matches that will be accepted.
-        In a [0, 1] interval. 1 means that only a perfect match will be
-        accepted, and zero means that the closest match will be accepted,
-        no matter how distant.
-        For numerical joins, this defines the maximum Euclidean distance
-        between the matches.
-    analyzer : {'word', 'char', 'char_wb'}, default=`char_wb`
-        Analyzer parameter for the CountVectorizer used for
-        the string similarities.
-        Describes whether the matrix `V` to factorize should be made of
-        word counts or character n-gram counts.
-        Option `char_wb` creates character n-grams only from text inside word
-        boundaries; n-grams at the edges of words are padded with space.
-    ngram_range : 2-tuple of int, default=(2, 4)
-        The lower and upper boundaries of the range of n-values for different
-         n-grams used in the string similarity. All values of `n` such
-         that ``min_n <= n <= max_n`` will be used.
+    max_dist : float, default=np.inf
+        Maximum acceptable (rescaled) distance between a row in the
+        ``main_table`` and its nearest neighbor in the ``aux_table``. Rows that
+        are farther apart are not considered to match. By default, the distance
+        is rescaled so that a value between 0 and 1 is typically a good choice,
+        although rescaled distances can be greater than 1 for some choices of
+        ``ref_dist``. ``None``, ``"inf"``, ``float("inf")`` or ``numpy.inf``
+        mean that no matches are rejected.
+    ref_dist : reference distance for rescaling, default = 'aux_percentile'
+        To facilitate the choice of ``max_dist``, distances between rows in
+        ``main_table`` and their nearest neighbor in ``aux_table`` will be
+        rescaled by this reference distance.
+    string_encoder : scikit-learn transformer used to vectorize text columns
```

Comment:
I'm not 100% sure myself, I kept the previous behavior in this regard. I would guess that the GapEncoder adds a lot of computation perhaps without much benefit here, that the TFIDF distances might be a bit closer to what people usually think about when considering string similarities, which is often tied to the number of ngrams in common, and that the minhashencoder is not adapted because distances between minhash outputs aren't really meaningful. If that sounds about right I'll add a note

Reply:
Sounds good!